File: Core/Stateflows.Common/StateMachines/Events/CurrentStateNotification.cs
using Newtonsoft.Json;
using Stateflows.Common;
using System.Collections.Generic;

namespace Stateflows.StateMachines.Events
{
    public sealed class CurrentStateNotification : BehaviorStatusNotification
    {
        [JsonProperty(TypeNameHandling = TypeNameHandling.None)]
        public IEnumerable<string> StatesStack { get; set; }
    }
}

File: Transport/@stateflows/common/src/events/current-state.notification.ts
import { Notification } from "./notification";

export class CurrentStateNotification extends Notification {
    constructor(
        public statesStack: Array<string>,
        public expectedEvents: Array<string>,
    ) {
        super();
    }

    public static notificationName = "Stateflows.StateMachines.Events.CurrentStateNotification";
}